modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -293,6 +295,32 @@ public ObjectRelationship deleteObjectRelationship(
 				"Reverse object relationships cannot be deleted");
 		}
 
+		List<Long> rows = objectRelationshipPersistence.dslQuery(
+			DSLQueryFactoryUtil.selectDistinct(
+				ObjectFieldSettingTable.INSTANCE.objectFieldId
+			).from(
+				ObjectRelationshipTable.INSTANCE
+			).innerJoinON(
+				ObjectFieldTable.INSTANCE,
+				ObjectFieldTable.INSTANCE.objectDefinitionId.eq(
+					ObjectRelationshipTable.INSTANCE.objectDefinitionId1)
+			).innerJoinON(
+				ObjectFieldSettingTable.INSTANCE,
+				ObjectFieldSettingTable.INSTANCE.objectFieldId.eq(
+					ObjectFieldTable.INSTANCE.objectFieldId)
+			).where(
+				ObjectRelationshipTable.INSTANCE.objectRelationshipId.eq(
+					objectRelationship.getObjectRelationshipId()
+				).and(
+					ObjectRelationshipTable.INSTANCE.name.eq(
+						ObjectFieldSettingTable.INSTANCE.value)
+				)
+			));
+
+		for (long id : rows) {
```

Comment:
Hey @G-Sa, could you please check if we can do here something like the code below?
We have two examples of this [here](https://github.com/liferay/liferay-portal/blob/a9f8768be3abb827ef90530adc23c3314ae21908/modules/apps/layout/layout-service/src/main/java/com/liferay/layout/internal/upgrade/v1_2_3/LayoutRevisionUpgradeProcess.java#L106-L119) and [here](https://github.com/liferay/liferay-portal/blob/a9f8768be3abb827ef90530adc23c3314ae21908/modules/apps/layout/layout-service/src/main/java/com/liferay/layout/internal/upgrade/v1_2_3/LayoutRevisionUpgradeProcess.java#L63-L83).

```

for (long objectFieldId : objectRelationshipPersistence.dslQuery(
		DSLQueryFactoryUtil.selectDistinct(
			ObjectFieldSettingTable.INSTANCE.objectFieldId
		).from(
			ObjectRelationshipTable.INSTANCE
		).innerJoinON(
			ObjectFieldTable.INSTANCE,
			ObjectFieldTable.INSTANCE.objectDefinitionId.eq(
				ObjectRelationshipTable.INSTANCE.objectDefinitionId1)
		).innerJoinON(
			ObjectFieldSettingTable.INSTANCE,
			ObjectFieldSettingTable.INSTANCE.objectFieldId.eq(
				ObjectFieldTable.INSTANCE.objectFieldId)
		).where(
			ObjectRelationshipTable.INSTANCE.objectRelationshipId.eq(
				objectRelationship.getObjectRelationshipId()
			).and(
				ObjectRelationshipTable.INSTANCE.name.eq(
					ObjectFieldSettingTable.INSTANCE.value)
			)
		))) {

	_objectFieldLocalService.deleteObjectField(objectFieldId);
}
```